src/base/castlerandom.pas

```diff
@@ -47,11 +47,12 @@   TCastleRandom = class(TObject)
     function RandomBoolean: boolean;
     { Randomly provides "-1", "0" or "1" with equal chances. }
     function RandomSign: longint;
-    { Returns a random number in 0 .. High(LongWord) range. }
+    { Returns a random number in 1 .. High(LongWord) range. }
```

Comment:
> RandomInt64 never returns 0

with the last fix it can return 0. Because we return `seed64bit mod N` which can obviously be 0 even in case `N` is `High(QWord)`.
However: `Random32bit` indeed can't be zero. To provide for zero randoms we have to upgrade xorshift algorithm with variable-cycling, e.g. like this one: https://en.wikipedia.org/wiki/Xorshift#Example_implementation
But on the other hand it reduces the speed.
So, basically I'd trade off RandomBoolean 5e-10 (0.00000005%) accuracy loss for speed :) I don't think anything like that might be ever significant. And in case that matters (e.g. like real-money casino), it's highly NOT advised to use any pseudorandom algorithms and revert to /dev/random (not even /dev/urandom) for truly random numbers :)